Step1 다각형 넓이 구하기

KaKaFE opened the pull request:
요구사항에 맞게 만들어보았습니다
피드백 부탁드립니다!

Commit:
step1 다각형 넓이 구하기 커밋입니다.

Commit:
Step2 1-1 요구사항 수정

Commit:
Step2 1-2 요구사항

printExecutionSequence 함수를 만들어 로깅을 구현,
실행 횟수기록 기능도 추가하였습니다

crongro commented:
드디어 step1 이라는 브랜치로 제대로 하신분이 나오셨네요!! 감사드리옵니다.
💯

crongro commented on index.js:
```diff
@@ -1,50 +1,96 @@
+}
+// 3.사다리꼴의 넓이
+function trapezoidSizecalc(base1, base2, height) {
+    if (arguments.length < 3) {
```
오류체크를 잘 했고요.
참고로, 이번 미션은 그렇지만, 늘 arguments.length를 확인하진 않아요.

crongro commented on index.js:
```diff
@@ -1,50 +1,96 @@
+function trapezoidSizecalc(base1, base2, height) {
+    if (arguments.length < 3) {
+        console.log('밑변,윗변의 길이와 높이를 입력해주세요')
+    } else if (typeof base1 !== 'number' || typeof base2 !== 'number' || typeof height !== 'number') {
```
이런식으로 if문안의 조건표현을 함수를 쓸 수 있을거 같아요. 

else if (isNumbers(arguments)) {}

crongro commented on index.js:
```diff
@@ -1,50 +1,96 @@
+    } else if (base1 <= 0 || base2 <= 0 || height <= 0) {
+        console.log('밑변과 윗변과 높이는 0보다 커야합니다')
+    } else {
+        var result = (base1 + base2) / 2 * height;
```
const / let을 쓰세요~

crongro commented on index.js:
```diff
@@ -1,50 +1,96 @@
+        console.log('반지름과 높이를 입력해주세요')
+    } else if (typeof radius !== 'number' || typeof height !== 'number') {
+        console.log('숫자만 입력해주세요')
+    } else if (radius <= 0 || height <= 0) {
```
엄격한 검사를 하셨네요.
이런코드가 늘어나니 메인로직은 숨겨져 있고 조건문이 더  중요해 보이긴하네요 ㅎ 

음 여기까지 체크하는 부분을 모두 모아서, 
별도 함수로 한번 빼볼래요?

crongro reviewed (changes requested):
꼼꼼한 체크가 돋보이는 코드네요.

리뷰남겨드렸듯이 에러체크 로직을 별도 함수로 한군데라도 한번 빼보실래요? 
(도전~!)

Commit:
step2 요구사항 수정

요구사항에 맞게 수정해보았습니다 피드백 부탁드립니다!
에러체크 로직을 최대한 밖으로 빼서 작동되게 노력해봤습니다

crongro commented on index.js:
```diff
@@ -9,75 +9,105 @@ function pushLog(shape) {
+    return false;
+}
+// 인자값이 있는지 확인하는 함수
+function isUndefined() {
```
여기서도 물론 arguments 를 쓸 수 있지만, 파라미터를 정의하고 쓰면 더 좋겠습니다.
이미 isUndefined()함수를 부르는 쪽에서 인자를 넘겨주기 때문에요.

crongro commented on index.js:
```diff
@@ -9,75 +9,105 @@ function pushLog(shape) {
-    } else if (arguments.length === 1) {
-        var result = Math.PI * Math.pow(radius1, 2);
+function circleSizecalc(radius1) {
+    if (errorCheck(radius1)) {
```
이렇게 if문 안에서 함수를 노출하면 (함수이름이 잘 지었다고 가정하고) 금방 무엇을 확인하는지 알기 쉽죠.

crongro commented on index.js:
```diff
@@ -9,75 +9,105 @@ function pushLog(shape) {
+// 1-1. 반지름을 입력받아 radius2값 만큼 계산
+function circlesSizecalc(radius1, radius2) {
+    if (errorCheck(radius1, radius2)) {
+        return;
```
잘 했어요. 빨리 반환할 거면 빨리 반환하는 거죠.

crongro commented on index.js:
```diff
@@ -9,75 +9,105 @@ function pushLog(shape) {
-        console.log('숫자만 입력해주세요')
-    } else if (base <= 0 || height <= 0) {
-        console.log('밑변과 높이는 0보다 커야합니다')
+    if (errorCheck(base, height)) {
```
약간 애매한데.. errorCheck는 무엇을 체크하는지 모르겠어요. 
checkError도 아니고요....

crongro commented on index.js:
```diff
@@ -9,75 +9,105 @@ function pushLog(shape) {
     }
 }
 // 4. 원기둥의 겉넓이 계산
 function CylinderSizecalc(radius, height) {
```
함수이름을 대문자로 시작하는건 생성자라는 걸로 취급하는 경우라서 잘 안씁니다.
소문자로 시작하세요~

crongro commented on index.js:
```diff
@@ -9,75 +9,105 @@ function pushLog(shape) {
+}
+// 숫자 확인 함수
+function isNumber() {
+    let len = arguments.length;
```
네 length값 비리 계산한건 잘했어요.

crongro reviewed (changes requested):
잘 수정했어요.
좀 아쉬운 부분이 있네요. 
한번더 수정해주세요~

Commit:
step2-1 수정사항 입니다

파라미터 정의와 몇몇 함수이름 변경을 하였으며,대소문자 구분을 했습니다
도저히 생각이 않나서 다른분들껄 참고해서 코드를 짯는데
다음부턴 혼자힘으로 최대한 하겠습니다!!

crongro reviewed (approved):
오 잘 수정하셨네요 👍 
머지합니다.
